openstudioreleases/src/auth/Profile.tsx

```diff
@@ -0,0 +1,281 @@
+import { FormikProvider, useFormik } from 'formik';
+import * as yup from 'yup';
+import {
+  type User,
+  updateProfile,
+} from 'firebase/auth';
+import { firebaseAuth, firebaseFirestore } from '../firebase';
+import { enqueueSnackbar } from 'notistack';
+import { useNavigate } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import { deleteDoc, doc, getDoc, updateDoc } from 'firebase/firestore';
+import { usePageView } from '../ga/usePageView';
+import { Header } from '../Header';
+import { enqueuePeristentErrorSnackbar } from '../utils/Snackbars'
+
+const getDocument = async (user_uid: string) => {
+
+  let values: UserValues = await getDoc(doc(firebaseFirestore, 'users', user_uid))
+  .then((docSnap) => {
+    // console.log("Inside the then");
+    if (docSnap.exists()) {
+      // console.log("docSnap.data()", docSnap.data());
+      return {
+        ...userValueDefaults,
+        ...docSnap.data(),
+      };
+    } else {
+      // docSnap.data() will be undefined in this case
+      // console.log(`No such document: users.${user_uid}`);
+      return userValueDefaults;
+    }
+  })
+  .catch((error) => {
+    // console.error(`No such document: users.${user_uid}`, error);
+    return userValueDefaults;
+  });
+
+  // console.log("values", values);
+  return values
+};
+
+
+export const Profile = () => {
+  usePageView();
+  const navigate = useNavigate();
+
+  if (!firebaseAuth.currentUser) {
+    return null;
+  }
+  const user: User = firebaseAuth.currentUser;
+
+  const [initialValues, setInitialValues] = useState<UserValues>(userValueDefaults);
+  const [initCountryLabelState, setInitCountryLabelState] = useState(null);
+  async function fetchInitialValues() {
+     const values = await getDocument(user.uid);
+     setInitialValues(values);
+     setInitCountryLabelState(CountryFromLabel(values.country));
+  }
+
+  useEffect(() => {
+    fetchInitialValues()
+  }, [])
+
+  const onSubmit = async (values: UserValues) => {
+    try {
+      const { firstName, lastName, country, company, occupation, email, signMailingList, joinBetaTester } = values;
+
+      await updateProfile(user, {
+        displayName: `${firstName} ${lastName}`
+      });
+
+      await updateDoc(doc(firebaseFirestore, 'users', user.uid), {
+        firstName,
+        lastName,
+        email,
+        company,
+        country,
+        occupation,
+        signMailingList,
+        joinBetaTester,
+      });
+
+      enqueueSnackbar('User Profile Updated', { variant: 'success' });
+
+    } catch (error) {
+      console.log(error);
+      enqueuePeristentErrorSnackbar(`Error Updating user profile: ${error}`);
+    }
+  };
+
+  const [openDeleteDialog, setOpenDeleteDialog] = useState(false);
+
+  const handleDeleteDialogClickOpen = () => {
+    setOpenDeleteDialog(true);
+  };
+
+  const handleDeleteDialogClose = () => {
+    setOpenDeleteDialog(false);
+  };
+
+  const onDeleteAccountSubmit = async () => {
```

Comment:
I haven't been able to get the delete to work even when I log out, log in, and hit delete as soon as I can

Reply:
I was able to delete the account when browsing to 127.0.0.1 but not localhost, not sure why.  After the account was deleted, the page gave this error.  Maybe we should call `window.close()` or redirect back to the create account page after deleting?

```
react-dom.production.min.js:188  Error: Minified React error #300; visit https://reactjs.org/docs/error-decoder.html?invariant=300 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at of (react-dom.production.min.js:160:337)
    at af (react-dom.production.min.js:196:258)
    at l (react-dom.production.min.js:291:88)
    at sB (react-dom.production.min.js:279:389)
    at react-dom.production.min.js:279:320
    at sM (react-dom.production.min.js:279:325)
    at sO (react-dom.production.min.js:270:88)
    at sI (react-dom.production.min.js:272:300)
    at r3 (react-dom.production.min.js:127:105)
    at react-dom.production.min.js:266:273
```